### PR TITLE
 (PA-2385) Update service management to run always for windows

### PR DESCRIPTION
**Pull request by McdonaldSeanp**

### CODE REVIEW

File: files/install_puppet.ps1 [hunk 1]
```diff
@@ -30,14 +30,17 @@ param(
   # correctly from the module.
   [parameter(Position=0)]
   [String] $PuppetPID,
-  [String] $Source='<%= @_msi_location %>',
-  [String] $Logfile='<%= @_logfile %>',
-  [String] $InstallDir='<%= @install_dir %>',
-  [String] $PuppetMaster='<%= @_puppet_master %>',
-  [String] $PuppetStartType='<%= @_agent_startup_mode %>',
-  [String] $InstallArgs='<%= @_install_options   %>',
-  [String] $InstallScriptPIDFile='<%= @_install_pid_file_loc %>',
```
Comment (speedofdark) [inline files/install_puppet.ps1]: This isn't in the PS based set of args? Is that on purpose? From reading below, it looks like you removed the need for it, but wanted to double check.

File: lib/puppet/parser/functions/windows_msi_installargs.rb [hunk 2]
```diff
@@ -14,6 +14,10 @@ module Puppet::Parser::Functions
         option
       end
     end
-    return arg_string.join(' ')
+    # When the MSI installargs parameter is passed to the powershell script it's inside
+    # a cmd.exe instance, so we need to escape the quotes correctly so they show up as
+    # plaintext double quotes to the powershell command. (To correctly escape to a
+    # plaintext " you use three "'s in cmd.exe)
```
Comment (speedofdark) [inline lib/puppet/parser/functions/windows_msi_installargs.rb]: I hate cross boundary quoting. A lot.

File: manifests/windows/install.pp [hunk 3]
```diff
@@ -29,25 +27,40 @@
...
     ensure  => file,
-    content => template('puppet_agent/install_puppet.ps1.erb')
+    content => file('puppet_agent/install_puppet.ps1')
   }
   -> exec { 'install_puppet.ps1':
-    command => "${::system32}\\cmd.exe /c start /b ${::system32}\\WindowsPowerShell\\v1.0\\powershell.exe -ExecutionPolicy Bypass -NoProfile -NoLogo -NonInteractive -File ${_installps1} ${::puppet_agent_pid}",
+    # The powershell execution uses -Command and not -File because -File will interpolate the quotes
+    # in a context like cmd.exe: https://docs.microsoft.com/en-us/powershell/scripting/components/console/powershell.exe-command-line-help?view=powershell-6#-file--
+    # Because of this it's much cleaner to use -Command and use single quotes for each powershell param
+    command => "${::system32}\\cmd.exe /S /c start /b ${::system32}\\WindowsPowerShell\\v1.0\\powershell.exe \
```
Comment (speedofdark) [inline manifests/windows/install.pp]: Our community will truly appreciate your commentary.

File: files/install_puppet.ps1 (hunk 1 above)
Comment (McdonaldSeanp) [reply files/install_puppet.ps1]: yeah, there's no need for this param anymore

**Pull request merged**